src/extender.js

```diff
@@ -0,0 +1,28 @@
+import shimmer from 'shimmer';
+import * as logger from './logger';
+import { safeExecute } from './utils';
+
+const noop = () => {};
+
+const isFunctionAlreadyWrapped = fn => fn && fn.__wrapped;
+
+export const hook = (module, funcName, options = {}) => {
+  const { beforeHook = noop, afterHook = noop } = options;
+  const safeBeforeHook = safeExecute(beforeHook, `before hook of ${funcName} fail`);
+  const safeAfterHook = safeExecute(afterHook, `after hook of ${funcName} fail`);
+  const extenderContext = {};
+  try {
+    const wrapper = originalFn => {
+      if (isFunctionAlreadyWrapped(originalFn)) return originalFn;
```

Comment:
don't you want to do this logic outside the wrapper, and then doesn't wrap?

Reply:
Its harder, because shimmer gives you access to the function, I don't want to access it directly